src/generators/date/index.ts

```diff
@@ -13,9 +13,15 @@ const createDateGenerator = (chance: Chance.Chance) => (
   from?: Date | string,
   to?: Date | string,
 ): Date => {
+  // https://tc39.es/ecma262/#sec-time-values-and-time-range
+  // https://tc39.es/ecma262/#_ref_6257:~:text=A%20Number%20can%20exactly%20represent%20all,beginning%20of%2001%20January%2C%201970%20UTC.
+  const minTimeMilliseconds = -8640000000000000;
+  const maxTimeMilliseconds = 8640000000000000;
+  const integer = createIntegerGenerator(chance);
+
   if (from == null) {
-    // https://github.com/goodeggs/avocado/commit/143f03ff267766be62189f901a174b02006650eb
-    from = 'Tue Oct 04 2011 10:44:00 GMT-0700 (PDT)';
+    const fromTime = integer(minTimeMilliseconds, maxTimeMilliseconds);
```

Comment:
I've read this code over and over and I'm having a hard time understanding the point of this. Wouldn't it be the same result but simpler code to just assign `minTimeMilliseconds` here rather than generate a random min that will later be used as the lower bound for another random int? I feel like this is unnecessarily indirect.

To restate the spec for the whole function:
- generate a valid Date between min and max, both of which are optional

If we just default min to `minTimeMilliseconds` and max to `maxTimeMilliseconds`, then generate a random date between min and max, doesn't that cover everything?
- `date()` → random date between `minTimeMilliseconds` and `maxTimeMilliseconds` (always valid by definition)
- `date(min)` → random date between `min` and `maxTimeMilliseconds` (always valid by definition)
- `date(null, max)` → random date between `minTimeMilliseconds` and `max` (always valid by definition)
- `date(min, max)` → random date between `min` and `max` (always valid if user passes in `min < max`)

Reply:
@serhalp you are absolutely right. Somehow my brain shut down on this one.
Addressed in b21b5d6